cli/src/cmds/loads/load.rs

```diff
@@ -179,64 +184,23 @@ impl LoadCommand {
                 let start = time::Instant::now();
                 let status = Status::read(self.conf.clone())?;
-                let (cli, url) = build_query_endpoint(&status)?;
-                let mut count = 0;
-                for _ in 0..args.value_of("skip_header_lines").unwrap().parse().unwrap() {
-                    record.next();
-                }
-                loop {
-                    let mut batch = vec![];
-                    // possible optimization is to run iterator in parallel
-                    for _ in 0..100_000 {
-                        if let Some(line) = record.next() {
-                            if let Ok(line) = line {
-                                batch.push(line);
-                                count += 1;
-                            } else {
-                                writer.write_err(format!(
-                                    "cannot read csv line {}, error: {}",
-                                    count,
-                                    line.unwrap_err()
-                                ))
-                            }
-                        } else {
-                            break;
-                        }
-                    }
-                    if batch.is_empty() {
-                        break;
-                    }
-                    let values: Vec<String> = batch
-                        .into_iter()
-                        .map(|s| {
-                            s.iter()
-                                .map(|i| {
-                                    if i.trim().is_empty() {
-                                        "null".to_string()
-                                    } else {
-                                        "'".to_owned() + i + &*"'".to_owned()
-                                    }
-                                })
-                                .join(",")
-                        })
-                        .map(|e| format!("({})", e.trim()))
-                        .filter(|e| !e.trim().is_empty())
-                        .collect();
-                    let values = values.join(",");
-                    let query = format!("INSERT INTO {} VALUES {}", table_format, values);
-                    if let Err(e) = execute_query_json(&cli, &url, query.clone()).await {
-                        writer.write_err(format!(
-                            "cannot insert data into {} with query {}, error: {:?}",
-                            table, query, e
-                        ))
-                    }
-                }
+                let (cli, url) = build_load_endpoint(&status)?;
+                let mut headers = HeaderMap::new();
+                headers.insert(
+                    "insert_sql",
+                    format!("INSERT INTO {} format CSV", table_with_schema)
+                        .parse()
+                        .unwrap(),
+                );
+                headers.insert("csv_header", "1".to_string().parse().unwrap());
```

Comment:
Skipping csv header need to be configurable through some flags, —skip-header-lines <num_line>